### DISCUSSION TITLE
希望为记忆倍速（使用倍速扩展后）增加黑名单机制

### DISCUSSION THREAD

**Discussion by beoxgame** [the1812/Bilibili-Evolved#4402] [功能建议 / Ideas]
倍速主要用处应该就是压缩说话慢吞吞的up主，提高信息密度节省观看时间。

同一up主的视频通常会保持固定的信息密度（如老蒋就会自己跳切，适合1.5倍速），而不同up主之间肯定会有很大差别（如龙女之声就适合2.0|2.5倍速）。

因此我觉得需要新增一个判定优先级高于默认全局倍速的倍速黑名单（`Dictionary<up主, 倍速值>`），这样在清动态视频（一次性观看当前订阅的up主发布的所有视频）时，能省去来回切换倍速的操作。~~一次两三秒，一天四五次，四舍五入就是一个亿啊~~

在UI交互上我觉得应该把倍速黑名单的操作区别于默认倍速，放在专门的入口中（因为通常设置一次就不会变了），然后在倍速列表里弄个样式（简单的颜色区分就行了），将黑名单倍速和全局倍速区分开来。
同时倍速列表里的切换强制固定为临时设置，不会影响黑名单倍速（以及全局倍速？）。

**Comment by Geeyun-JY3** [comment]
应该是重复：#1315

**Reply by beoxgame** [reply]
差不多，但我想的是拿`up主id`作为key，那个讨论是`视频编号`作为key（应该就是现在记忆倍速里的那个“各视频分别记忆”了）

**Reply by Geeyun-JY3** [reply]
确实搞错了，应该是：#3738

**Comment by ckyuex** [comment]
我觉得你直接装html5插件就好了

**Reply by beoxgame** [reply]
你说的h5插件是什么？Bilibili-Evolved的插件在线仓库搜不出来

**Reply by ckyuex** [reply]
https://h5player.anzz.top/
